module3-crud-and-data-models/r3-model-view-controller/README.md

```diff
@@ -0,0 +1,83 @@
+# Model View Controller (MVC)
+
+## Introduction to MVC projects
+
+When beginning a new project, there are several different methods that can be used to set up the overall file structure and flow. One of the most commonly used architectural patterns is called MVC. This is an acronym for `Model`, `View`, `Controller`.
+This pattern is favored because of its alignment with the computer science design principle, Separation of Concerns. By dividing up responsibilities within our file structure, we can encapsulate information to be referred to via abstraction and maintain cleaner codebases.
+
+```
+app-name-here
+ |- Controllers
+ |- db
+ |- Models
+ |- node_modules
+.gitignore
+index.js
+package.json
+package-lock.json
+```
+
+### Models
+
+In the models folder, the files will contain our schema definition and expected fields for the model.
+The naming convention for the model files is: `name-here-model.js`.
+
+Here’s an example model using JavaScript and mongoose.
+
+```js
+// post-model.js
+
+const mongoose = require("mongoose");
+
+const post = new mongoose.Schema({
+  title: {
+    type: String,
+    required: true,
+  },
+  body: {
+    type: String,
+    required: true,
+  },
+  createdAt: {
+    type: Date(),
+  },
+  author: {
+    type: String,
+    required: true,
+  },
+});
+
+module.exports = mongoose.model("Post", post);
+```
+
+### Controllers
+
+Controllers are the layer between the Model and the View. the views can use the controllers to `add`, `read`, `delete`, ...etc data.
```

Comment:
i think this explanation needs to be much better of what a controller is.

Reply:
it is worth mentioning that the controller is where you validate and sanitize the data sent to be persisted in the database.